roles/gather/tasks/facts.yaml

```diff
@@ -19,8 +15,29 @@
   ansible.builtin.debug:
     msg: "{{ gather_result }}"
 
+# Generate reports in specified formats
+- name: Set default formats if not specified
+  ansible.builtin.set_fact:
+    format: ["yaml"]
+  when: format is undefined
+
+- name: Create reports directory
```

Comment:
why are we creating directories? for gather we just need to register these into variables